### PR TITLE
Replaced fwlinks in csharp\programming-guide\generics folder

### PR DESCRIPTION
**Pull request by pkulikov**
Replaced fwlinks in **csharp\programming-guide\generics** folder.
One link was broken, so I've found a couple of articles from the docs as replacement.
Related to #3424 

Moreover, replaced .NET Framework mention with .NET. Strictly speaking, that makes this PR contain changes of two kinds. Is it fine for such 'refinement' changes or it's better to create separate PRs in future?

### CODE REVIEW

**Comment by mairaw** [comment]
Hi @pkulikov, usually it's better if the PR focus on one task unless we're totally revisiting the topics.

We do have a separate issue (#2028) where we're tracking the work to review the topics to not be Framework and Windows centric.

File: docs/csharp/programming-guide/generics/benefits-of-generics.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ ms.author: "wiwagn"
 # Benefits of Generics (C# Programming Guide)
 Generics provide the solution to a limitation in earlier versions of the common language runtime and the C# language in which generalization is accomplished by casting types to and from the universal base type <xref:System.Object>. By creating a generic class, you can create a collection that is type-safe at compile-time.  
   
- The limitations of using non-generic collection classes can be demonstrated by writing a short program that uses the <xref:System.Collections.ArrayList> collection class from the .NET Framework class library. <xref:System.Collections.ArrayList> is a highly convenient collection class that can be used without modification to store any reference or value type.  
+ The limitations of using non-generic collection classes can be demonstrated by writing a short program that uses the <xref:System.Collections.ArrayList> collection class from the .NET class library. <xref:System.Collections.ArrayList> is a highly convenient collection class that can be used without modification to store any reference or value type.
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/generics/benefits-of-generics.md]: Although it's not part of your PR's intent, @pkulikov, would you mind deleting the phrase "is a highly convenient collection class that"? It sounds suspiciously like a recommendation.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for both revising this topic for multiple .NET implementations, @pkulikov, and for revising the links. Would you mind making one more change that is unrelated to your PR? Then this will be ready to merge.

File: docs/csharp/programming-guide/generics/benefits-of-generics.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/programming-guide/generics/benefits-of-generics.md]: @rpetrusha that's true, I've removed that one and then realized that "can be used without modification" also sounds a little bit like a recommendation. So, I've removed that one as well. Please review the change.

**Comment by pkulikov** [comment]
@mairaw thanks for the link to that issue. I'll keep my PRs more one task centered.
@rpetrusha I've implemented requested change, please review it.

**Review by rpetrusha — APPROVED** [review]
I really appreciate your making the additional change, @pkulikov. This looks really good. We'll merge your PR now.

**Pull request merged**